File: rest-api/src/main/java/com/fuelstation/managmentapi/fuelstation/domain/FuelStationService.java
package com.fuelstation.managmentapi.fuelstation.domain;

import com.fuelstation.managmentapi.common.domain.FuelGrade;
import com.fuelstation.managmentapi.fuelstation.domain.models.FuelStation;

public interface FuelStationService {
    FuelStation createFuelStation(String street, String buildingNumber, String city, String postalCode, String country);
    FuelStation assignManager(long fuelStationId, long managerId);
    FuelStation unassignManager(long fuelStationId, long managerId);
    FuelStation changeFuelPrice(long fuelStationId, FuelGrade fuelGrade, float newPrice);
    FuelStation processFuelDelivery(long fuelStationId, long fuelOrderId);
    FuelStation deactivateFuelStation(long fuelStationId);
}

File: rest-api/src/main/kotlin/com/fuelstation/managmentapi/fuelstation/domain/FuelStationService.kt
package com.fuelstation.managmentapi.fuelstation.domain

import com.fuelstation.managmentapi.common.domain.FuelGrade

interface FuelStationService {
    fun createFuelStation(street: String, buildinNumber: String, city: String, postalCode: String, country: String)
    fun assignManager(gasStationId: Int, managerId: Int)
    fun unassignManager(gasStationId: Int, managerId: Int)
    fun changeFuelPrice(gasStationId: Int, fuelGrade: FuelGrade, newPrice: Float)
    fun processFuelDelivery(fuelOrderId: Int)
}